code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -306,25 +235,61 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
 	stored_paper--
 	return TRUE
 
-/obj/item/modular_computer/InsertID(obj/item/inserting_item)
-	var/obj/item/computer_hardware/card_slot/card_slot = all_components[MC_CARD]
-	var/obj/item/computer_hardware/card_slot/card_slot2 = all_components[MC_CARD2]
-
-	if(!(card_slot || card_slot2))
+/**
+ * InsertID
+ * Attempt to insert the ID in either card slot.
+ * Args:
+ * inserting_id - the ID being inserted
+ * user - The person inserting the ID
+ */
+/obj/item/modular_computer/InsertID(obj/item/card/inserting_id, mob/user)
+	//all slots taken
+	if(computer_id_slot)
 		return FALSE
 
-	var/obj/item/card/inserting_id = inserting_item.GetID()
-	if(!inserting_id)
-		return FALSE
+	computer_id_slot = inserting_id
+	if(user)
+		if(!user.transferItemToLoc(inserting_id, src))
+			return FALSE
+		to_chat(user, span_notice("You insert \the [inserting_id] into the card slot."))
+	else
+		inserting_id.forceMove(src)
 
-	if((card_slot?.try_insert(inserting_id)) || (card_slot2?.try_insert(inserting_id)))
-		if(ishuman(loc))
-			var/mob/living/carbon/human/human_wearer = loc
-			if(human_wearer.wear_id == src)
-				human_wearer.sec_hud_set_ID()
-		update_appearance()
-		update_slot_icon()
+	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
+	if(ishuman(loc))
+		var/mob/living/carbon/human/human_wearer = loc
+		if(human_wearer.wear_id == src)
+			human_wearer.sec_hud_set_ID()
+	update_appearance()
+	update_slot_icon()
+	return TRUE
+
+/**
+ * Removes the ID card from the computer, and puts it in loc's hand if it's a mob
+ * Args:
+ * user - The mob trying to remove the ID, if there is one
+ */
+/obj/item/modular_computer/RemoveID(mob/user)
+	if(!computer_id_slot)
+		return ..()
+
+	if(user)
+		if(!issilicon(user) && in_range(src, user))
```

Comment:
you sure it works correctly? in the video the id card pops on the ground rather than in hands.

Reply:
![image](https://user-images.githubusercontent.com/53777086/202888239-1dc50340-3fef-45f5-992e-5a8492e427e4.png)

I had fixed it and tested it after I had made the video